src/build/env.py

```diff
@@ -126,6 +134,9 @@ def install(self, requirements):  # type: (Iterable[str]) -> None
         if not requirements:
             return
 
+        env = dict(os.environ)
+        env['PIP_NO_WARN_SCRIP_LOCATION'] = '1'
```

Comment:
```suggestion
        env['PIP_NO_WARN_SCRIPT_LOCATION'] = '1'
```

Reply:
I suppose we can revert to using the flag now, actually.

Reply:
😳

Reply:
When this passes, I'll update the changelog and test name.

Reply:
We could, but this is safer, just in case something goes wrong. But I can drop this commit if desired.

Reply:
Enough is passing, will repush

Reply:
This might also need to be set to `'0'` for it to have the desired effect, see https://github.com/pypa/pip/issues/5200.

Reply:
I've been bit by the oddities of NO variables in pip before; would be nice if there was a good way to test this. Maybe by looking for the warning?

Reply:
IMHO we can use the flag here 👍🏻

Reply:
> I've been bit by the oddities of NO variables in pip before; would be nice if there was a good way to test this. Maybe by looking for the warning?

That would require figuring out how to trigger the warning.  If pip uses the same Python executable as the script we're installing, pip does not generate a warning; see https://github.com/pypa/pip/blob/c7419b2aac16c1190b274b9cd7c273d879816ddb/src/pip/_internal/operations/install/wheel.py#L178.

Reply:
I've made this `'0'`. I think, since this is just hiding a warning, the environment variable is a bit better. I would not want to risk breaking just to try to hide a warning. If we require pip 10+ (personally I like 18.1+ due to manylinux2010 support :) ) somewhere else, that's fine, but it probably would be better to break there.

I've still kept this a distinct commit so it's trivial to drop.